src/traces/sankey/defaults.js

```diff
@@ -83,11 +83,11 @@ module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout
     };
 
     if(traceOut.node.label.some(missing)) {
-        Lib.log('Some of the nodes are neither sources nor targets, please remove them.');
+        Lib.warn('Some of the nodes are neither sources nor targets, please remove them.');
```

Comment:
Minor point, the user may have different ways to solve the problem (eg providing the missing links, instead of removing disconnected nodes). I would have the warning simply describe what we did and why, like "Some of the nodes are neither sources nor targets, they will not be displayed." (if indeed that's what will happen) Your message on encountering cycles does a great job of this.